elyra/pipeline/processor_local.py

```diff
@@ -0,0 +1,130 @@
+        for operation in operations:
+            notebook = self.get_absolute_path(operation.filename)
+            if not os.path.isfile(notebook):
+                raise FileNotFoundError(f'Could not find {notebook}')
+
+            self._execute_file(pipeline_work_dir, notebook)
+
+        return PipelineProcessorResponse('', '', object_storage_path=pipeline_work_dir)
+
+    def export(self, pipeline, pipeline_export_format, pipeline_export_path, overwrite):
+        raise NotImplementedError('Local pipelines does not support export functionality')
+
+    def _execute_file(self, work_dir, filename):
+        notebook_dir = os.path.dirname(filename)
+        notebook_name = os.path.basename(filename)
+        stdout = os.path.join(work_dir, f'{notebook_name}.out')
+        stderr = os.path.join(work_dir, f'{notebook_name}.err')
+
+        self.log.debug(f'Processing: {filename}')
+        with open(stdout, 'w') as stdout_file, open(stderr, 'w') as stderr_file:
+            t0 = time.time()
+            try:
+                # Try executing with configured python kernel
+                papermill.execute_notebook(
+                    input_path=filename,
+                    output_path=filename,
+                    stdout_file=stdout_file,
+                    stderr_file=stderr_file,
+                    cwd=notebook_dir
+                )
+            except KeyError:
+                # force default python kernel
+                papermill.execute_notebook(
+                    input_path=filename,
+                    output_path=filename,
+                    stdout_file=stdout_file,
+                    stderr_file=stderr_file,
+                    cwd=notebook_dir,
+                    kernel_name="python3"
+                )
+            except Exception:
+                self.log.error(f'Internal error executing {filename}')
+
+            t1 = time.time()
+            duration = (t1 - t0)
+            self.log.debug(f'Execution of {notebook_name} took {duration:.3f} secs.')
+
+    @staticmethod
+    def _get_pipeline_work_dir(pipeline_name: str) -> str:
+        log_dir = os.path.join(tempfile.tempdir, 'elyra', pipeline_name)
+        os.makedirs(log_dir, mode=0o700, exist_ok=True)
+        return log_dir
+
+    @staticmethod
+    def _get_operations_by_dependency(operations_by_id: dict) -> list:
+        ordered_operations = []
+
+        for operation in operations_by_id.values():
+            if operation not in ordered_operations:
+                # operation is a root node
+                if not operation.parent_operations:
+                    ordered_operations.append(operation)
+                else:
+                    LocalPipelineProcessor.\
+                        _visit_operation(operations_by_id, ordered_operations, operation)
+
+        return ordered_operations
+
+    @staticmethod
+    def _visit_operation(operations_by_id, ordered_operations, operation):
```

Comment:
This seems more than a 'visit'.  I would argue its `_sort_operation`.
Nonetheless, please add some comments and a help string.

Could you add type hints here?  You might also need to pull things like `Dict` and `List` from the `typing` package if you need to use `Optional[]`, etc.